src/Command/BackUpCommand.php

```diff
@@ -55,170 +31,150 @@ protected function execute(InputInterface $input, OutputInterface $output)
+                $site_list[] = (object) ['name' => $environment->openshiftProjectName, 'routes' => $environment->routes];
             }
-            $alias_file_content .= $single_alias;
         }
 
         fwrite($alias_file, $alias_file_content);
 
         print("\nCopying file [" . $destination . "govcms.aliases.drushrc.php] to [/home/govcms/.drush/govcms.aliases.drushrc.php]");
+        mkdir('/home/govcms/.drush/', 0777, true);
         copy($destination . "govcms.aliases.drushrc.php", "/home/govcms/.drush/govcms.aliases.drushrc.php");
         $list_of_files = array();
 
         foreach ($site_list as $site) {
             $temp_count++;
             $start = time();
             print "\n***************************\n";
-            print "Starting Backup of " . $site->site . " [" . $site->domains[0] . "] #" . $temp_count . "/" . sizeof($site_list) . "\n";
-            exec("drush @" . $site->domains[0] . " ssh 'mkdir /mnt/tmp/backups;chmod 777 /mnt/tmp/backups'");
-            exec("drush @" . $site->domains[0] . " archive-dump --destination=/mnt/tmp/backups/" . $site->domains[0] . ".tar.gz --overwrite  --tar-options=\"--exclude=sites/default/files/* --exclude=sites/default/files.bak/* --exclude=sites/default/files-private/* --exclude=sites/default/private-files/*\"");
+            print "Starting Backup of " . $site->name . " #" . $temp_count . "/" . sizeof($site_list) . "\n";
+            exec("drush @" . $site->name . " archive-dump --destination=/tmp/dr-backups/" . $site->name . ".tar.gz --overwrite  --tar-options=\"--exclude=sites/default/files/* --exclude=sites/default/files.bak/* --exclude=sites/default/files/private/* \"");
             print "Dump completed.\n";
-            mkdir($destination . $site->domains[0]);
-            print "Retrieving " . $site->site . " [" . $site->domains[0] . "] dump.\n";
-            exec("drush -y rsync --remove-source-files @" . $site->domains[0] . ":/mnt/tmp/backups/" . $site->domains[0] . ".tar.gz " . $destination . $site->domains[0] . "/ > /dev/null 2>/dev/null &");
-            $list_of_files[] = $site->domains[0] . ".tar.gz";
-            $domains = $site->domains[0];
-            if (isset($site->collection_domains) && !empty($site->collection_domains)) {
-                $domains = implode(" ", $site->collection_domains);
-                $domains .= " " . $site->domains[0];
-            }
-            $sites_file_content .= $site->domains[0] . " " . $site->domains[0] . ".tar.gz " . $destination . $site->domains[0] . "/" . $site->domains[0] . ".tar.gz " . $site->id . " \"" . $domains . "\"\n";
+            mkdir($destination . $site->name);
+            print "Retrieving " . $site->name . " dump.\n";
+            exec("drush -y rsync --remove-source-files @" . $site->name . ":/tmp/dr-backups/" . $site->name . ".tar.gz " . $destination . $site->name . "/ ");
```

Comment:
@paulkilla 
It seems the previous version ran the `rsync` in the background and then had a `sleep(120);` later.
This seems rather dangerous to me, as it could mean that the `rsync` is not finished and the backup fails.
What do you think?

Reply:
@Schnitzel It was setup so that the rsync occured in the background while other tasks completed.
Happy to remove it and do it non-async... can at least test it and see how it goes.